### PR TITLE
rust: impl TryFrom<SBP> for all messages

### PR DESCRIPTION
**Pull request by notoriaga**
Still working on the event handler thing, but it definitely still needs some work. I think for now it should live in the console as it's a pretty good proving ground for the concept. this is just the minimum changes that are needed to implement it elsewhere. Both changes also seem generally useful

### CODE REVIEW

File: generator/sbpg/targets/resources/sbp_messages_mod.rs [hunk 1]
```diff
@@ -37,6 +37,21 @@ pub trait SBPMessage: SbpSerialize {
     }
 }
 
+pub trait RealMessage: SBPMessage + std::convert::TryFrom<SBP, Error = TryFromSBPError> + Clone + Sized {
+    const MESSAGE_TYPE: u16;
```
Comment (notoriaga) [inline generator/sbpg/targets/resources/sbp_messages_mod.rs:41]: Could also add `const MESSAGE_NAME: &static str` here as well I suppose

File: generator/sbpg/targets/resources/sbp_messages_mod.rs [hunk 2]
```diff
@@ -37,6 +37,21 @@ pub trait SBPMessage: SbpSerialize {
     }
 }
 
+pub trait RealMessage: SBPMessage + std::convert::TryFrom<SBP, Error = TryFromSBPError> + Clone + Sized {
```
Comment (notoriaga) [inline generator/sbpg/targets/resources/sbp_messages_mod.rs:40]: Not sure if `RealMessage` is the best name, but it's what it's called in the code generator

Comment (silverjam) [reply generator/sbpg/targets/resources/sbp_messages_mod.rs:40]: `ConcreteMessage`? `KnownMessage`? `ReifiedMessage`?

File: generator/sbpg/targets/resources/sbp_messages_mod.rs (hunk 1 above)
Comment (silverjam) [reply generator/sbpg/targets/resources/sbp_messages_mod.rs:41]: 👍

**Review by silverjam — COMMENTED** [review]
lgtm, I think adding `MESSAGE_NAME` makes sense and I like `ConcreteMessage` (but I'll defer to whatever name you prefer)

**Review by silverjam — APPROVED** [review]

**Review by jbangelo — APPROVED** [review]

**Pull request merged**